arangod/Replication2/StateMachines/Document/DocumentLeaderState.cpp

```diff
@@ -64,7 +64,10 @@ DocumentLeaderState::DocumentLeaderState(
           _handlersFactory->createSnapshotHandler(core->getVocbase(), gid)),
       _errorHandler(_handlersFactory->createErrorHandler(gid)),
       _guardedData(std::move(core), _handlersFactory),
-      _transactionManager(transactionManager) {}
+      _transactionManager(transactionManager) {
+  // Get ready to replay the log
+  _shardHandler->prepareShardsForLogReplay();
```

Comment:
NIT
This can be moved to `DocumentLeaderState::recoverEntries()` - that's the place where the leader may replay the log (unlike the follower). I think that might be a bit cleaner, but surely, the constructor must happen before that, so feel free to keep it here if you it better fits your needs.

Reply:
I would like to keep it in the Constructor.
Just to have little difference between leader and follower and to make it clear for anyone looking at this in the future that this operation is required for both, and required round and about the same time.